### DISCUSSION TITLE
in the grid export different columns than seen in the grid

### DISCUSSION THREAD

**Discussion by teafmartins** [serenity-is/Serenity#6851] [Help]
Hello,
How in a grid can I export a different set of columns than what is seen?
In the grid I only want to see 3 customer columns, but when I click to export I want to export 10 more customer columns.

I am using the following code in the endpoint to export the columns:

        public FileContentResult ListExcel(IDbConnection connection, ListRequest request)
        {
            var data = List(connection, request).Entities;
            var report = new DynamicDataReport(data, request.IncludeColumns, typeof(Columns.CustomerColumns));
            var bytes = new ReportRepository().Render(report);
            return ExcelContentResult.Create(bytes, "CustomerList_" +
                DateTime.Now.ToString("yyyyMMdd_HHmmss") + ".xlsx");
        }

**Comment by wezmag** [comment]
Try add the columns you need to `request.IncludeColumns`, before call `List(connection, request)`.
```c#
public FileContentResult ListExcel(IDbConnection connection, ListRequest request)
{
    request.IncludeColumns.AddRange(new string[] {
        nameof(Columns.CustomerColumns.FirstColumn),
        nameof(Columns.CustomerColumns.SecondColumn),
        nameof(Columns.CustomerColumns.ThirdColumn),
        // .... more columns
    });
    var data = List(connection, request).Entities;
    var report = new DynamicDataReport(data, request.IncludeColumns, typeof(Columns.CustomerColumns));
    var bytes = new ReportRepository().Render(report);
    return ExcelContentResult.Create(bytes, "CustomerList_" +
        DateTime.Now.ToString("yyyyMMdd_HHmmss") + ".xlsx");
}
```
